log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/MdcSerializer.java

```diff
@@ -24,27 +24,42 @@
  */
 package co.elastic.logging.log4j2;
 
+import java.util.ServiceLoader;
+
 import co.elastic.logging.EcsJsonSerializer;
 import co.elastic.logging.JsonUtils;
 import org.apache.logging.log4j.core.LogEvent;
 import org.apache.logging.log4j.util.TriConsumer;
 
-interface MdcSerializer {
+public interface MdcSerializer {
```

Comment:
```suggestion
interface DefaultMdcSerializer extends MdcSerilaizer {
```

I'd suggest to rename this class to `DefaultMdcSerializer` and instead introduce a new public interface `MdcSerializer` which defines just the contract to be implemented by users wanting to customize the behaviour.

Reply:
Yes, but now the initialization looks like this:  
`private static final MdcSerializer MDC_SERIALIZER = DefaultMdcSerializer.Resolver.resolve();`.  

To me, it feels a bit odd that the default implementation handles the resolution. Plus, I won’t be able to do this during the class initialization phase - it has to be done in the constructor since I’m using the `ecslayout` parameter.  

I’m thinking of adding a new interface and moving the resolution logic into it. What do you think?